contrib/inventory/digital_ocean.py

```diff
@@ -449,18 +461,22 @@ def build_inventory(self):
 
             self.inventory['all']['hosts'].append(dest)
 
-            self.inventory[droplet['id']] = [dest]
-            self.inventory[droplet['name']] = [dest]
+            if droplet['id'] not in self.inventory:
+                self.add_inventory_group(droplet['id'])
+            self.inventory[droplet['id']]['hosts'].append(dest)
+
+            if droplet['name'] not in self.inventory:
+                self.add_inventory_group(droplet['name'])
+            self.inventory[droplet['name']]['hosts'].append(dest)
 
             # groups that are always present
-            for group in ('digital_ocean',
-                          'region_' + droplet['region']['slug'],
+            for group in ('region_' + droplet['region']['slug'],
                           'image_' + str(droplet['image']['id']),
                           'size_' + droplet['size']['slug'],
                           'distro_' + DigitalOceanInventory.to_safe(droplet['image']['distribution']),
                           'status_' + droplet['status']):
                 if group not in self.inventory:
-                    self.inventory[group] = {'hosts': [], 'vars': {}}
+                    self.add_inventory_group(group)
```

Comment:
This block is the same in the 3 cases, and assuming that we can use add_hosts, should we simple move this condition to function add_host?
Then this can replaced by:
```
self.add_host(group, dest) ?
```

Reply:
Your referencing the entire if condition that is checking if the group is already in the inventory, correct?

Reply:
In this example, moving `if group not in self.inventory` into the `add_host` call and performing the entire operation within that single method call. Just making sure I'm following given I was more focused on the document that was being sent to stdout.

Reply:
@BondAnthony  yes the the entire if condition. No big deal tho.